### PR TITLE
Switch from hiccup to clostache templating

### PR DESCRIPTION
**Pull request by johnworth**
Using Clostache to templatize the download pages instead of generating them with hiccup will make future changes to kifshare's ui a bit easier to accomplish. This should not change anything major in the appearance of the UI and should not affect performance.

### CODE REVIEW

File: src/kifshare/ui_template.clj [hunk 1]
```diff
@@ -1,221 +1,32 @@
...
-
-  (html
-   [:div {:id "size-detail"}
-    [:div {:id "size-label"}
-     [:p "File Size:"]]
-    [:div {:id "size"}
-     [:p (FileUtils/byteCountToDisplaySize
-          (Long/parseLong (:filesize ticket-info)))]]]))
+(def memo-read-template
+  (memoize read-template))
```
Comment (psarando) [inline src/kifshare/ui_template.clj:13]: `read-template` isn't getting called anywhere else, so I would push `memoize` down into the `read-template` function and get rid of `memo-read-template`.

File: ui/ui.xml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.4.2/underscore-min.js" type="text/javascript"></script>
+    <script src="resources/js/jquery.tooltip.min.js" type="text/javascript"></script>
+    <script src="resources/js/jquery.zclip.min.js" type="text/javascript"></script>
+    <script src="resources/js/json2.js" type="text/javascript"></script>
+    <script src="resources/js/kif.js" type="text/javascript"></script>
+  </head>
+
+  <body>
+    <span id="ticket-info" style="display: none">
+      <div id="ticket-info-map">{{{ticket-info-json}}}</div>
```
Comment (ianmcorvidae) [inline ui/ui.xml:22]: Just to double-check: does mustache/clostache support variable names with hyphens? A lot of stuff is stupid about that, but I'm assuming you tested.

**Review by psarando — APPROVED** [review]
I like this approach over hiccup. I didn't compare the new XML to the old hiccup templates, so if nothing essential got lost in the conversion, then I say :shipit:

**Review by ianmcorvidae — APPROVED** [review]
Nothing sticks out as a problem, just one question about stuff I don't myself know.

File: ui/ui.xml (hunk 2 above)
Comment (johnworth) [reply ui/ui.xml:22]: Yep, I checked it out. It's working fine.

File: src/kifshare/ui_template.clj (hunk 1 above)
Comment (johnworth) [reply src/kifshare/ui_template.clj:13]: I'm working on adding file-watching for the template, so this stuff will probably get changed later today anyway.

**Pull request merged**